fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java

```diff
@@ -338,4 +1100,118 @@ private TableName qualifiedNameToTableName(QualifiedName qualifiedName) {
+                    }
+                } else if (signature.equalsIgnoreCase("CHAR")) {
+                    if (type.typeParameter().size() > 1) {
+                        throw new SemanticException("CHAR can not contains multi type parameter");
+                    }
+                    if (type.typeParameter(0).INTEGER_VALUE() != null) {
+                        return ScalarType.createCharType(
+                                Integer.parseInt(type.typeParameter(0).INTEGER_VALUE().toString()));
+                    } else {
+                        throw new SemanticException("CHAR type parameter mush be integer");
+                    }
+                } else if (signature.equalsIgnoreCase("DECIMAL")) {
+                    throw new IllegalArgumentException("Unsupported type specification: " + type.getText());
+                }
+
+                throw new IllegalArgumentException("Unsupported type specification: " + type.getText());
+            }
+
+            if (signature.equalsIgnoreCase("BOOLEAN")) {
+                return Type.BOOLEAN;
+            } else if (signature.equalsIgnoreCase("TINYINT")) {
+                return Type.TINYINT;
+            } else if (signature.equalsIgnoreCase("SMALLINT")) {
+                return Type.SMALLINT;
+            } else if (signature.equalsIgnoreCase("INT") || signature.equalsIgnoreCase("INTEGER")) {
+                return Type.INT;
+            } else if (signature.equalsIgnoreCase("BIGINT")) {
+                return Type.BIGINT;
+            } else if (signature.equalsIgnoreCase("LARGEINT")) {
+                return Type.LARGEINT;
+            } else if (signature.equalsIgnoreCase("FLOAT")) {
+                return Type.FLOAT;
+            } else if (signature.equalsIgnoreCase("DOUBLE")) {
+                return Type.DOUBLE;
+            } else if (signature.equalsIgnoreCase("DECIMAL")) {
+                return ScalarType.createUnifiedDecimalType(10, 0);
+            } else if (signature.equalsIgnoreCase("DATE")) {
+                return Type.DATE;
+            } else if (signature.equalsIgnoreCase("DATETIME")) {
+                return Type.DATETIME;
+            } else if (signature.equalsIgnoreCase("TIME")) {
+                return Type.TIME;
+            } else if (signature.equalsIgnoreCase("VARCHAR")) {
+                return Type.VARCHAR;
+            } else if (signature.equalsIgnoreCase("CHAR")) {
+                return Type.CHAR;
+            } else if (signature.equalsIgnoreCase("STRING")) {
+                ScalarType stringType = ScalarType.createVarcharType(ScalarType.DEFAULT_STRING_LENGTH);
+                stringType.setAssignedStrLenInColDefinition();
+                return stringType;
+            } else if (signature.equalsIgnoreCase("BITMAP")) {
+                return Type.BITMAP;
+            } else if (signature.equalsIgnoreCase("HLL")) {
+                return Type.HLL;
+            } else if (signature.equalsIgnoreCase("PERCENTILE")) {
```

Comment:
JSON type is ignored

Reply:
How about move this method to Type.java, to make the code of type system more compact?

Reply:
> JSON type is ignored

When I was developing, JSON TYPE was also under development. So it is not included. I don't want to introduce JSON TYPE support in this PR anymore. I can support JSON TYPE in the next separate PR.

Reply:
> How about move this method to Type.java, to make the code of type system more compact?

I totally agree with this idea. But there's too much confusion in Type at the moment, and I don't think now is the right time to make it more complicated. For example, the `public static ScalarType createType(String type) ` function in ScalarType generates Type from a string, but its definition is not clear and cannot meet the actual needs of Parser. So I don't want to abstract a type generator function that is only used in Parser into Type in this PR.